replace.js

```diff
@@ -0,0 +1,43 @@
+const isSpecialRegex = /(\$\$)|(\$&)|(\$`)|(\$')|(\$\d)/
+function getReplace (addon) {
+  if (addon.replace) {
+    return addon.replace
+  }
+  return function replace (transactionId, result, thisArg, matcher, replacer) {
+    if (transactionId && typeof thisArg === 'string' && typeof replacer === 'string') {
+      if (typeof matcher === 'string') {
```

Comment:
Encapsulate following lines in a method to avoid duplication?
```
if (addon.isTainted(transactionId, thisArg, replacer)) {
  const isSpecialReplacement = replacer.indexOf('$') > -1 && !!replacer.match(isSpecialRegex)
  if (!isSpecialReplacement) {
  }
}
```

Reply:
done